### PR TITLE
Modularization

### PR DESCRIPTION
**Pull request by europaul**

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -10,6 +10,6 @@ FROM alpine:edge
 RUN apk add --no-cache bash
 SHELL ["/bin/bash", "-c"]
 COPY --from=builder /app/app ./
-COPY ./docker/entrypoint.sh /entrypoint.sh
+COPY ./docker/docker-entrypoint.sh /entrypoint.sh
 
 ENTRYPOINT ["/entrypoint.sh"]
```
Comment (shriftz) [inline docker/Dockerfile:15]: entrypoint.sh used twice above but but the file is deleted

File: docker/Dockerfile [hunk 2]
```diff
@@ -10,6 +10,6 @@ FROM alpine:edge
 RUN apk add --no-cache bash
 SHELL ["/bin/bash", "-c"]
 COPY --from=builder /app/app ./
-COPY ./docker/entrypoint.sh /entrypoint.sh
+COPY ./docker/docker-entrypoint.sh /entrypoint.sh
```
Comment (shriftz) [inline docker/Dockerfile:13]: docker-entrypoint.sh is not used

File: .github/workflows/docker_publish.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  build-push-image:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Log in to the Container registry
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
```
Comment (shriftz) [inline .github/workflows/docker_publish.yml:17]: secrets are coming from git workflow?

File: makefile [hunk 4]
```diff
@@ -1,6 +1,22 @@
...
+	docker run -p 80:80 --rm --env-file=./.env --entrypoint /bin/bash -it ${MODULE}:${VERSION_NAME}
+.phony: debug_image
+
+push_latest:
+	docker image push ${MODULE}:${VERSION_NAME}
+.phony: push_latest
+
 create_and_push_multi_platform:
-	docker buildx build --platform linux/amd64,linux/arm64,linux/arm/v6,linux/arm/v7 -t ${MODULE}:${VERSION_NAME} --push . -f docker/Dockerfile
+	docker buildx build --platform linux/amd64,linux/arm64,linux/arm/v6,linux/arm/v7 -t ${MODULE}:${VERSION_NAME} . -f docker/Dockerfile
```
Comment (shriftz) [inline makefile]: if push is removed, images will be in cache only

File: makefile [hunk 5]
```diff
@@ -1,6 +1,22 @@
...
+	docker run -p 80:80 --rm --env-file=./.env ${MODULE}:${VERSION_NAME}
+.phony: run_image
+
+debug_image:
+	docker run -p 80:80 --rm --env-file=./.env --entrypoint /bin/bash -it ${MODULE}:${VERSION_NAME}
+.phony: debug_image
+
+push_latest:
+	docker image push ${MODULE}:${VERSION_NAME}
+.phony: push_latest
```
Comment (shriftz) [inline makefile]: pushing single image for current arch?

File: src/processor/process.go [hunk 6]
```diff
@@ -25,34 +26,41 @@ func Process(msg models.Data) ([]byte, error) {
 func validateData(msg models.Data) error {
```
Comment (shriftz) [inline src/processor/process.go]: This validation can be done same as you did in Agent using validation package

File: docker/Dockerfile (hunk 1 above)
Comment (europaul) [reply docker/Dockerfile:15]: in line 13 we copy the file `docker-entrypoint.sh` from the `docker` dir to the image and call it `entrypoint.sh` there

File: docker/Dockerfile (hunk 2 above)
Comment (europaul) [reply docker/Dockerfile:13]: it is used in line 13

File: .github/workflows/docker_publish.yml (hunk 3 above)
Comment (europaul) [reply .github/workflows/docker_publish.yml:17]: exactly

File: makefile (hunk 4 above)
Comment (europaul) [reply makefile]: thanks, nice catch!

File: makefile (hunk 5 above)
Comment (europaul) [reply makefile]: this will probably never be used. I'll remove it

File: docker/Dockerfile (hunk 1 above)
Comment (shriftz) [reply docker/Dockerfile:15]: oh my bad, I thought we copying two different files there

**Review by shriftz — APPROVED** [review]
Nice changes, specially module standard files

**Pull request merged**